### PR TITLE
feat(plugin, cli, serve-web): Add conversation browser web UI plugin

### PR DESCRIPTION
**Pull request by JeanMertz**
The new `jp-serve-web` plugin serves a read-only web interface for browsing JP conversations. It integrates with the command plugin system and communicates with the `jp` host over the JSON-lines plugin protocol (stdin/stdout).

Run it via `jp serve web`, or with custom options:

    jp serve web --bind 0.0.0.0 --port 8080

The server defaults to `127.0.0.1:3000`. Bind and port can also be set in `.jp/config.toml` under `[plugins.command.serve.options]`.

The plugin fetches conversation data on demand through the host protocol and renders it as HTML using `maud` templates, with `axum` as the HTTP server. Markdown is rendered via `comrak`. A custom tracing layer buffers log events during startup, then forwards them to the host as `PluginToHost::Log` messages once the connection is established.

### CODE REVIEW

File: crates/plugins/command/serve-web/src/render.rs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    serde_json::to_string_pretty(val).unwrap_or_else(|_| val.to_string())
+}
+
+fn truncate(s: &str, max: usize) -> String {
+    if s.len() <= max {
+        s.to_owned()
+    } else {
+        let truncated: String = s.chars().take(max).collect();
+        format!("{truncated}\n\n... (truncated)")
+    }
```
[the remark this answers is not in the document]
Comment (JeanMertz) [reply crates/plugins/command/serve-web/src/render.rs]: Fixed. `truncate` now works in a single unit: it cuts at `max` bytes, walking back to the nearest char boundary, so the guard and the cut agree and no spurious `... (truncated)` is appended to content that fits. Added a multibyte test covering the boundary case.

File: crates/plugins/command/serve-web/src/routes.rs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    State(state): State<AppState>,
+    Path(id): Path<String>,
+) -> Result<Markup, AppError> {
+    debug!(%id, "GET /conversations/{{id}}");
+
+    let resp = state
+        .client
+        .read_events(&id)
+        .await
+        .map_err(|e| AppError::NotFound(e.to_string()))?;
```
[the remark this answers is not in the document]
Comment (JeanMertz) [reply crates/plugins/command/serve-web/src/routes.rs]: Fixed. `read_events` errors now map `ClientError::Host(_)` → `AppError::NotFound` (the host's signal for a missing conversation) and every other variant → `AppError::Internal`, so `ChannelClosed`/protocol errors are 500s. `NotFound` renders a fixed "Conversation not found." message instead of the raw error string, and the host message is logged server-side with `debug!` rather than leaked to the client.

File: crates/plugins/command/serve-web/src/routes.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+    let mut headers = axum::http::HeaderMap::new();
+    headers.insert(
+        header::CONTENT_TYPE,
+        HeaderValue::from_static("text/css; charset=utf-8"),
+    );
+    headers.insert(
+        header::CACHE_CONTROL,
+        HeaderValue::from_static("public, max-age=31536000, immutable"),
+    );
```
[the remark this answers is not in the document]
Comment (JeanMertz) [reply crates/plugins/command/serve-web/src/routes.rs]: Fixed by versioning the URL, with one tweak to the suggested snippet: `css_etag()` returns the hash already wrapped in quotes, so using it directly in `?v=` would produce `?v=%22…%22`. I split out a `css_version()` that returns the bare hex; the stylesheet link is now `href=(format!("/assets/style.css?v={}", style::css_version()))` and `css_etag()` wraps `css_version()` for the header. A CSS change produces a new URL, so `immutable` is correct.

File: crates/plugins/command/serve-web/src/render.rs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+/// Convert markdown to HTML using comrak.
+pub(crate) fn markdown_to_html(md: &str) -> String {
+    let mut options = comrak::Options::default();
+    options.render.r#unsafe = true;
```
[the remark this answers is not in the document]
Comment (JeanMertz) [reply crates/plugins/command/serve-web/src/render.rs]: Fixed. Dropped `options.render.r#unsafe = true`, so comrak escapes/omits raw HTML while still rendering markdown, which is all this read-only viewer needs. Added a regression test asserting `<script>` doesn't survive into the output while surrounding markdown still renders. If raw HTML pass-through is ever wanted, that's a follow-up with `ammonia` sanitization before `PreEscaped`.
